projects/pt1/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py

```diff
@@ -412,6 +412,23 @@ def aten〇sum〇dim_IntList〡shape(self: List[int], dim: Optional[List[int]],
 def aten〇prod〇dim_int〡shape(self: List[int], dim: int, keepdim: bool = False, dtype: Optional[int] = None) -> List[int]:
     return upstream_shape_functions.sum_mean_dim(self, [dim], keepdim, dtype)
 
+def aten〇pixel_shuffle〡shape(self: List[int], upscale_factor: int) -> List[int]:
+
+    assert len(self) >= 3, "input must be at least rank-3 in pixel_shuffle"
+    upscale_factor_squared = upscale_factor * upscale_factor
+    assert self[-3] % (upscale_factor_squared) == 0, "number of input channels  must be divisible by upscale_factor^2 in pixel_shuffle"
+    
+    out = self[0:-3]
+    out.append(self[-3] // upscale_factor_squared)
+    out.append(self[-2] * upscale_factor)
+    out.append(self[-1] * upscale_factor)
+    return out
+
+def aten〇pixel_shuffle〡dtype(self_rank_dtype: Tuple[int, int], upscale_factor: int) -> int:
```

Comment:
nit: all dtype functions here are tested e2e. Can you add a test to this one? It should be a matter of adding something like `@check_dtype_function(_check_tensors_with_the_same_dtype(num_of_tensors=1, upscale_factor=1))`

Reply:
You can run the tests by generating the shape and dtype library: `./build_tools/update_abstract_interp_lib.sh`

Reply:
Done!